### ISSUE TITLE
Make s3proxy proper express middleware

### ISSUE THREAD

**Issue by gmoon** [gmoon/s3proxy#13]
Something like this:

`var express = require('express')
var app = express()
var S3Proxy = require('s3proxy')
const s3proxy = new S3Proxy({ bucket: 's3proxy-public' })
proxy.init()

// load the s3proxy middleware
app.get('/*', s3proxy)`

**Comment by codeviking** [comment]
As a current user of the library, I actually prefer the API as is.  It allows us to embed additional logic surrounding the streaming proxy, i.e.

```javascript
app.get('/some/url', (req, res) => {
  fetchSomeThing()
     .then((exists) =>  {
          if (exists) {
              proxy.stream(req, res) 
          }  else {
             res.status(404).send()
         }
     });
});
```

That said, I can see why the middleware approach would be nice, as it's less setup for a new user.

Maybe you can retain both APIs? 

Thanks for the great library!

**Comment by gmoon** [comment]
I agree, the middleware approach is not well suited for the streaming nature of the current API. I will leave this open in case anyone has suggestions for this concept. Thanks for the feedback!
